### DISCUSSION TITLE
How to customize the tauri dev/build command

### DISCUSSION THREAD

**Discussion by joe20182** [tauri-apps/tauri#4964] [Q&A] [answered]
I have built the app with vite for frontend.
And in my `package.json` I have such scripts:
```json
{
  "scripts": {
    "dev": "vite --mode development",
    "build": "vite build",
}
```
I also set my `tauri.config.json`
```json
{
  "build": {
    "distDir": "../dist",
    "devPath": "http://localhost:3000",
    "beforeDevCommand": "yarn dev",
    "beforeBuildCommand": "yarn build"
  }
}
```
So that when I run `yarn tauri dev`, I can start the app in dev environment.
But is there a way to run in different environment with different command?
I'm think about running `yarn tauri dev --mode staging`(or `yarn tauri staging`) and it will trigger `vite --mode staging` as `beforeDevCommand`.
Is that possible? Thank you!

**Comment by FabianLars** [comment, marked as answer]
First of all, this is not _directly_ possible, but you can workaround it;
I think the easiest way would be to not use beforeDevCommand at all, and instead do it inside the package.json scripts via https://www.npmjs.com/package/concurrently or https://www.npmjs.com/package/npm-run-all.

Alternatively you can create a second tauri config which looks like this
```json
{
  "build": {
    "beforeDevCommand": "yarn dev --mode staging"
  }
}
```
And invoke it by using something like `yarn tauri dev -c staging.json`.


As a last resort you could just run the commands in 2 different terminals.

**Reply by joe20182** [reply]
I think creating a second tauri config is a good way, thank you!
